lib/model/CEventRateModel.cc

```diff
@@ -477,15 +476,23 @@ bool CEventRateModel::computeProbability(std::size_t pid,
     LOG_TRACE(<< "probability(" << this->personName(pid) << ") = " << p);
 
     resultBuilder.probability(p);
-    resultBuilder.anomalyScoreExplanation(pJoint.anomalyScoreExplanation());
 
     double multiBucketImpact{-1.0 * CAnomalyDetectorModelConfig::MAXIMUM_MULTI_BUCKET_IMPACT_MAGNITUDE};
     if (pJoint.calculateMultiBucketImpact(multiBucketImpact)) {
         resultBuilder.multiBucketImpact(multiBucketImpact);
     }
 
+    resultBuilder.anomalyScoreExplanation(pJoint.anomalyScoreExplanation());
     bool everSeenBefore = this->firstBucketTimes()[pid] != startTime;
-    resultBuilder.personFrequency(this->personFrequency(pid), everSeenBefore);
+    auto typicalConcentration = m_Probabilities.medianConcentration();
+    double actualConcentration;
+    if (m_ProbabilityPrior.concentration(pid, actualConcentration) &&
+        typicalConcentration.has_value()) {
```

Comment:
It seems more principled to test the simple condition first
```suggestion
    if (typicalConcentration.has_value() &&
        m_ProbabilityPrior.concentration(pid, actualConcentration)) {
```

Reply:
Also should we just grab `auto& explanation = resultBuilder.anomalyScoreExplanation();` upfront. Might make the code a little cleaner.

Reply:
changed